src/lib/core/Router.ts

```diff
@@ -0,0 +1,85 @@
+import RouteRecognizer from 'route-recognizer'
+import * as React from 'react'
+import { createMemoryHistory, Action, Location, LocationDescriptor, History } from 'history'
+import { RouteProps } from '../../index'
+
+export interface RouteConfig {
+  path: string
+  handler: React.ComponentType<RouteProps>
+}
+
+export type Match
+  = undefined
+  | MatchedRoute
+
+export interface MatchedRoute extends RouteProps<any, any> {
+  handler: React.ComponentType<any>
+}
+
+export class Router {
+  static match(location: LocationDescriptor, routes: RouteConfig[], action: Action): Match {
+    const router = new Router(routes, createMemoryHistory())
+    router.set(location, action)
+    return router.route
+  }
+
+  private recognizer = createRecognizer()
+  private history: History
+
+  constructor(routes: RouteConfig[], history: History) {
+    routes.forEach((route) => {
+      this.recognizer.add([route])
+    })
+
+    this.history = history
+  }
+
+  get route(): Match {
+    const matches = this.recognizer.recognize(this.path)
+    if (!matches) {
+      return undefined
+    }
+
+    const match = matches[0]
+
+    /* istanbul ignore next */
```

Comment:
Is this telling the coverage checker to ignore the no match = undefined case - if so, why's that?

Reply:
The istanbul ignore stuff is for the code coverage tool. Will document why it's being ignored!